install/operator/pkg/syndesis/configuration/configuration.go

```diff
@@ -291,7 +298,11 @@ func GetProperties(file string, ctx context.Context, client client.Client, synde
 		if err := client.Get(ctx, types.NamespacedName{Namespace: syndesis.Namespace, Name: "syndesis-db"}, databaseDeployment); err == nil {
 			for _, c := range databaseDeployment.Spec.Template.Spec.Containers {
 				if c.Name == "postgresql" {
-					configuration.DatabaseNeedsUpgrade = c.Image != configuration.Syndesis.Components.Database.Image
+					currentPostgreSQLVersion, err := postgresqlVersionFromImage(client, c.Image)
+					if err != nil {
+						return nil, err
```

Comment:
As a general point, we return and log errors like this, & for the most part these sort of errors never get thrown. However, on the rare instance it does, what do we expect to happen? Operators are supposed to be self-healing etc... so how much mitigation should we put in place over time to try and isolate and kill off any possible corner cases that may arise?

@heiko-braun @lgarciaaco

Reply:
Does an operator have a somehting like a preflight phase? I hitnk checks like these should be performed, before any modifications are made.